server/issue.go

```diff
@@ -867,49 +858,110 @@ func (s *Server) createDatabaseCreateTaskList(ctx context.Context, c api.CreateD
 	}, nil
 }
 
-// creates PITR TaskCreate list and dependency.
-func createPITRTaskList(database *api.Database, projectID int, targetTs int64) ([]api.TaskCreate, []api.TaskIndexDAG, error) {
+func (s *Server) createPITRTaskList(ctx context.Context, originDatabase *api.Database, projectID int, c api.PITRContext) ([]api.TaskCreate, []api.TaskIndexDAG, error) {
 	var taskCreateList []api.TaskCreate
 
-	// task: create and restore to PITR database
+	// We make sure that createPITRTaskList will always return 2 tasks.
+	taskIndexDAGList := []api.TaskIndexDAG{
+		{
+			FromIndex: 0,
+			ToIndex:   1,
+		},
+	}
+
+	// Restore payload
 	payloadRestore := api.TaskDatabasePITRRestorePayload{
-		ProjectID:     projectID,
-		PointInTimeTs: &targetTs,
+		ProjectID: projectID,
 	}
+
+	if c.CreateDatabaseCtx != nil {
+		targetInstance, err := s.store.GetInstanceByID(ctx, c.CreateDatabaseCtx.InstanceID)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to find the instance with ID: %d, err: %w", c.CreateDatabaseCtx.InstanceID, err)
+		}
+		project, err := s.store.GetProjectByID(ctx, projectID)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to find the project with ID: %d, err: %w", projectID, err)
+		}
+		taskList, err := s.createDatabaseCreateTaskList(ctx, *c.CreateDatabaseCtx, *targetInstance, *project)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to create the database create task list, error: %w", err)
+		}
+		taskCreateList = append(taskCreateList, taskList...)
+
+		payloadRestore.TargetInstanceID = &targetInstance.ID
+		payloadRestore.DatabaseName = &c.CreateDatabaseCtx.DatabaseName
+	}
+	if c.BackupID != nil {
+		// We only support restore the full backup to a new database only, so c.CreateDatabaseCtx must be not nil in this case.
```

Comment:
We will support in-place backup restore very soon, like this week :D

Please think of backup restore equal to pitr, and redesign the related code.